### ISSUE TITLE
Add support for signals without any parameters

### ISSUE THREAD

**Issue by Gravetow** [ssannandeji/Zenject-2019#412]
In other DI frameworks it is possible to declare and fire signals without any parameters (i'm most familiar with strange), of course this is not a necessary feature or anything.

But sometimes I just want to have a signal that is simply that - a signal. Not necessary including any parameters.

**Comment by johnlcox** [comment]
This is already possible. The no-args signal still uses a generic type parameter of the signal class itself. You would define you Signal as an empty class and provide your class as the generic type parameter. The `Fire` and `Listen` handlers would not take any parameters.

```C#
public class MyAwesomeSignal : Signal<MyAwesomeSignal>
{
}

// In your installer
Container.DelcareSignal<MyAwesomeSignal>();

// Firing it
_myAwesomeSignal.Fire();

// Listening
_myAwesomeSignal.Listen(HandleAwesomeSignal)
private void HandleAwesomeSignal()
{
  // Do awesome things.
}
```

**Comment by Gravetow** [comment]
Thank you!

I think I did try exactly your approach, yet I couldn't implement the _Listening_ part! It might have something to do with my setup, as I use the Unity 2018.1beta and Projeny.

I'll try again and in doubt post more info about the error here - maybe I stumbled upon a "bug". 

But it's more likely I just did something wrong somewhere haha~

**Comment by Gravetow** [comment]
So I tried the setup you proposed and still get an error shown from the compiler.

Turns out that in the ISignal Interface the Listen and Unlisten methods show errors in my solution.

Apparently there is no Action without any parameters defined. 

![missingaction](https://user-images.githubusercontent.com/8374080/36448629-742d74b6-1688-11e8-8a80-045293f07111.png)

UPDATE: 
I changed my .Net version in the Player preferences in Unity for testing purpose and changed it back again, suddenly it finds the needed Action.
Turning off and on again saves the day apparently ~
I'll close this then.
